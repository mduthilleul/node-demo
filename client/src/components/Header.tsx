//import reactLogo from "../assets/react.svg";
//import viteLogo from "/vite.svg";

export type HeaderProps = {
  title: string;
};

export const Header = ({ title }: HeaderProps) => {
  return (
    <>
      <div>
        {/*
        <a href="https://vite.dev" target="_blank">
          <img src={viteLogo} className="logo" alt="Vite logo" />
        </a>
        <a href="https://react.dev" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
        */}
      </div>
      <h1 data-testid="title">{title}</h1>
    </>
  );
};
