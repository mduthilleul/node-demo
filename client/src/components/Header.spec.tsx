import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Header } from "./Header";

describe("Header", () => {
  test("it should render the title with h1", async () => {
    render(<Header title="plop" />);
    const title = await screen.findByTestId("title");
    expect(title).toHaveTextContent("plop");
    expect(title.tagName).toEqual("H1");
  });
});
