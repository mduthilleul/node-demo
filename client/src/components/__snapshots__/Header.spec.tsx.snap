// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header it should match snapshot 1`] = `
<DocumentFragment>
  <div />
  <h1
    data-testid="title"
  >
    plop
  </h1>
</DocumentFragment>
`;
